src/persist-types/src/codec_impls.rs

```diff
@@ -611,52 +611,73 @@ impl ColumnCfg<Option<OpaqueData>> for () {
     }
 }
 
-impl ColumnRef<()> for Bitmap {
+impl ColumnRef<()> for BooleanBuffer {
     fn cfg(&self) -> &() {
         &()
     }
+
     fn len(&self) -> usize {
         self.len()
     }
-    fn to_arrow(&self) -> (Encoding, Box<dyn Array>) {
-        let array = BooleanArray::new(ArrowLogicalType::Boolean, self.clone(), None);
-        (Encoding::Plain, Box::new(array))
+
+    fn to_arrow(&self) -> Arc<dyn Array> {
+        Arc::new(BooleanArray::new(self.clone(), None))
     }
-    fn from_arrow(_cfg: &(), array: &Box<dyn Array>) -> Result<Self, String> {
+
+    fn from_arrow(_cfg: &(), array: &dyn Array) -> Result<Self, String> {
         let array = array
             .as_any()
             .downcast_ref::<BooleanArray>()
             .ok_or_else(|| format!("expected BooleanArray but was {:?}", array.data_type()))?;
-        if array.validity().is_some() {
+        if array.logical_nulls().is_some() {
             return Err("unexpected validity for non-optional bool".to_owned());
         }
+
         Ok(array.values().clone())
     }
 }
 
-impl ColumnGet<bool> for Bitmap {
+impl ColumnGet<bool> for BooleanBuffer {
     fn get<'a>(&'a self, idx: usize) -> bool {
-        self.get_bit(idx)
+        self.value(idx)
     }
 }
 
-impl ColumnPush<bool> for MutableBitmap {
+impl ColumnPush<bool> for BooleanBufferBuilder {
     fn push<'a>(&mut self, val: bool) {
-        <MutableBitmap>::push(self, val)
+        <BooleanBufferBuilder>::append(self, val)
+    }
+
+    fn finish(mut self) -> <bool as Data>::Col {
+        <BooleanBufferBuilder>::finish(&mut self)
+    }
+}
+
+impl ColumnMut<()> for BooleanBufferBuilder {
+    fn new(_cfg: &()) -> Self {
+        // Note(parkmycar): This capacity was picked arbitrarily.
+        BooleanBufferBuilder::new(128)
```

Comment:
Looks like `BooleanBuilder::default()` below ends up using `1024` for the capacity. We could make this match to be less arbitrary?

Reply:
Ahh great call! Let me update this